Add Runtime assemblies for VB Framework projects

basoundr opened the pull request:
VB Framework projects needs references to the Core libraries, without which Object and all the essential types will be missing. This reference was added by VisualBasic Project in the legacy world. This change will perform a similar functionality.

Tagging @srivatsn @dotnet/project-system for review

**Customer scenario**

Without this change, the language service for all of VB framework projects is broken

**Bugs this fixes:** 

#1961 

**Workarounds, if any**

None

**Risk**

Low. This change is only for VB framework projects.

**Performance impact**

None. There is no complex computation. 

**Is this a regression from a previous update?**

No. This scenario could not have worked before unless the Core libraries were added by mistake.

**Root cause analysis:**

This is parity behavior of the VB Legacy Project which is one of the many things that VB does.

**How was the bug found?**

Ad-hoc testing

srivatsn commented:
The VB compiler [injects these references](http://source.roslyn.io/#Microsoft.CodeAnalysis.VisualBasic/CommandLine/VisualBasicCommandLineParser.vb,1195) as well. The old VB compiler and language service used to inject these references. To maintain compat, vbc injects them as well - that's why commandline build is working in this case. Basically Roslyn chose to keep the compilation API clean and not have to know about this weird behavior and delegated it to the hosts. 

To fix this from the build side, we could do that in the SDK but then desktop projects would still be broken. For the old PS the VB projectystem shim in Roslyn that derives from AbstractProject implements this behavior - for CPS we only have CPSProject and no language specific components. Maybe we need a VBCPSProject on the Roslyn side and the factory would create based on language. That might also make it so that anycode for vb would get the right behavior.

srivatsn commented:
Dave and I discussed this and there's a better place to fix this - in the CompileDesignTime target. @basoundr let's discuss tomorrow morning.

basoundr commented:
@srivatsn @davkean I have made the changes to CompileDesignTime as per the discussion.

@dotnet/project-system for review

srivatsn commented on src/Targets/Microsoft.VisualBasic.DesignTime.targets:
```diff
@@ -31,21 +31,32 @@
+        >
+
+    <!-- VB Framework projects needs core library references. In the case of Command line scenario, the vbc injects this reference. -->
+    <PropertyGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'">
```
This check doesn't sound right - we should use the same logic that the legacy LS is using which I believe doesn't check the TFM but instead just checks if FrameworkPathOverride is non null.

srivatsn replied:
I believe it also checks if nostdlib is set or not.

basoundr replied:
@srivatsn Before adding the reference we check if the path exists, which would cover the `FrameworkPathOverride` check. This would also cover us from cases in the future, when we decide to set `FrameworkPathOverride`

srivatsn replied:
This would mean that PCLs are now broken they have a different TFI - what about xamarin? This is a rabbit hole you don't want to go down. Just replicating the same logic as the old LS and the compiler is the easiest and most consistent.

srivatsn commented on src/Targets/Microsoft.VisualBasic.DesignTime.targets:
```diff
@@ -31,21 +31,32 @@
+    <!-- VB Framework projects needs core library references. In the case of Command line scenario, the vbc injects this reference. -->
+    <PropertyGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'">
+      <MSCorlibPath>$(FrameworkPathOverride)\mscorlib.dll</MSCorlibPath>
+      <MSVBPath>$(FrameworkPathOverride)\Microsoft.VisualBasic.dll</MSVBPath>
```
Doesn't the old LS test for vbcore - the VBRuntime property - if that's set to Embed then we shouldn't be adding a reference.

basoundr commented:
@srivatsn I have addressed your concerns and I have combined the behavior of [Vbc Task](http://index/?leftProject=Microsoft.Build.Tasks.v4.0&leftSymbol=v0z2ullltq6z&file=Vbc.cs&line=430) and old LS behavior(http://index/?query=visualbasicprojectoptions&rightProject=Microsoft.VisualStudio.LanguageServices.VisualBasic&file=ProjectSystemShim%5CVisualBasicProjectOptionsHelper.vb&line=114)

basoundr commented:
@davkean Could you please take another look here?

basoundr commented:
@davkean If you have any more concerns with this approach, we can discuss it further.

davkean commented:
@basoundr No I'm fine with this approach - looks good.
